src/Plugin/Field/FieldWidget/AutomatedListingConfigurationWidget.php

```diff
@@ -46,13 +47,28 @@ class AutomatedListingConfigurationWidget extends StringTextareaWidget implement
    */
   protected $index;
 
+  /**
+   * @var ImmutableConfig
+   */
+  private $config;
```

Comment:
@asarfara cc @anthony-malkoun you should use the field widget configuration instead of using system configuration.

Reply:
Not a bad idea :)